### ISSUE TITLE
Switch to `pnpm workspaces` instead of `yarn + lerna`

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-ui#3102]
### Description

i just read this article here: https://www.raulmelo.dev/blog/replacing-lerna-and-yarn-with-pnpm-workspaces and found out that https://pnpm.io/workspaces can replace lerna + yarn.

Advantages would be:
- yarn workspaces and lerna seem to be not so much maintained
- one tool to combine everything
- pnpm offers a huge speed increase when installing / reinstalling packages
- pnpms  speed - seriously, this  weights at least twice ^^

Disadvantages:
- it currently works -> never change a running system
- people need to install pnpm and must not use yarn anymore
- i dont know how we can preserve the yarn.lock into a pnpm-lock.yaml (to preserve the dependencies for now *)

i just tried it out locally and this seems to be working good -> but we didn't do a `yarn upgrade` since some time, so the dependency `@friendsofreactjs/react-css-themr` makes worries as the ts declarations are not found (because they were moved to the `types` folder?).
Since that makes already trouble, i think its best to avoid changing the total install stack and upgrading the packages at once, right?

(so as i found no way to transform a yarn.lock to a pnpm-lock.yaml we would need to upgrade all deps to the latest version that is allowed in package.json and then switch to pnpm)

edit there is a native option: https://pnpm.io/cli/import was quite hard to fin xD

**Comment by Sebobo** [comment]
I think the writer of the blog post you mention only used Yarn 1 which was replaced by Yarn 2 quite a while ago. 
IMO Yarn 2 focuses a lot on the points you mention and gets regular releases.
I use the Yarn 1 workspace in several projects (especially the new media-ui) and I'm very happy with them.
But I didn't do so much with Yarn 2 features yet. I planned to try to replace the lerna stuff in the Neos.Ui at some point with the yarn workspace feature and also upgrade to Yarn 2 there.

Honestly performance of installing and reinstalling is the least problem in the js world for me. The biggest pain is all the config files and dependency stuff breaking all the time. pnpm doesn't solve that from what I understood.
That's what I like about the yarn workspaces is that the subpackages have no special config beside the normal package.json contents and the root package.json has a reference where the sub packages are located. That's all you need.

So should we in parallel try yarn 2 and replace lerna and see what works best?
The dependency issues should be the same for both approaches.

**Comment by mhsdesign** [comment]
im open to any approaches ^^ (and happy to help)
btw the first linked blog post is not so well written - i know xD this one explains things way better: https://divriots.com/blog/switching-to-pnpm
and it even says that yarn3 is nearly as fast as pnpm (look at the end) so this is would be a good option too? (its also explained how  they speeded up yarn ...)

**Comment by markusguenther** [comment]
Lerna is now official a dead project and we should replace it.

**Comment by mhsdesign** [comment]
uups yes funny that note happended withing the last 8 days https://github.com/lerna/lerna/

maybe yarn3/yarn2 + workspaces? - as far as i know we use yarn 1 with not so cool workspace support.

pnpm would we cool too but after putting some initial time into it its still not working: https://github.com/neos/neos-ui/pull/3103

**Comment by Sebobo** [comment]
Yarn workspaces work fine in all versions. I use it mostly still with yarn 1.
Lets try next week to get v2 basically running and see if we can replace lerna. Yarn 2 allows opt-in into some of the new features, so we can take step by step.

**Comment by mhsdesign** [comment]
closing in favor of: #3115 (https://github.com/neos/neos-ui/pull/3116)
